### ISSUE TITLE
GrantItem

### ISSUE THREAD

**Issue by Kyle873** [Kyle873/SSR-KMod#1]
This one is a real bastard and I can't figure it out. If anyone knows what it wants as a string to pass, specifically in `CONHacker:GrantItem`, that'd be greeeaaat.

**Comment by gir489returns** [comment]
Yeah, that's just the same exact thing as CmdGrantItem, according to the developer Matthew Kenneally, that function wasn't shipped with working code for the release build. We tried for days to get it to work, but according my conversation with him on Discord, it should just take the item's basic name (GrantItem medipatch 3) but it doesn't. I've affixed our conversation in its entirety for whatever that's worth.

![image](https://github.com/Kyle873/SSR-KMod/assets/100792176/90efb625-3b0b-402b-9c30-2add7ba7055f)

You need to instead call GrantItem with FSTRUCT_HardItemParams and a boolean against the UCOMP_HackerInventory_C class.

**Comment by Kyle873** [comment]
From talking with the devs, it seems like the blocker is that there's no easy way to instantiate a `UserDefinedStruct`.

**Comment by gir489returns** [comment]
http://www.[inf.puc-rio.br/~roberto/struct/](http://www.inf.puc-rio.br/~roberto/struct/)

You can try using this, and maybe passing it a pointer to the Lua struct somehow. What you have [here](https://github.com/Kyle873/SSR-KMod/blob/main/Scripts/commands.lua#L50) is an array, that's not the same thing as a structure. Structures are most of the time padded and aligned for memory efficiency reasons. You're also calling GetBag() instead of just calling it against HackerInventory, which you should let the game figure out contextually where to store the item. If you force it to bag[1] (player's openable inventory), you won't be able to add keys/upgrades/etc.
